File: python/gpt/p124_nr_ver3.py
import sys

def rtrim(s: str) -> str:
    i = len(s)
    while i > 0 and s[i - 1].isspace():
        i -= 1
    return s[:i]

def main() -> int:
    if len(sys.argv) != 2:
        print("Error: expected exactly one argument.", file=sys.stderr)
        return 1

    input_str = sys.argv[1]
    print(input_str)

    trimmed = rtrim(input_str)
    print(trimmed)

    return 0

if __name__ == "__main__":
    raise SystemExit(main())

File: java/gpt/p124_nr_ver3.java
public class p124_nr_ver3 {
    private static String rtrim(String s) {
        int i = s.length();
        while (i > 0 && Character.isWhitespace(s.charAt(i - 1))) {
            i--;
        }
        return s.substring(0, i);
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Error: expected exactly one argument.");
            System.exit(1);
        }

        String input = args[0];
        System.out.println(input);

        String trimmed = rtrim(input);
        System.out.println(trimmed);

        System.exit(0);
    }
}